lib/devise_saml_authenticatable/saml_config.rb

```diff
@@ -1,16 +1,39 @@
 require 'ruby-saml'
 module DeviseSamlAuthenticatable
   module SamlConfig
-    def saml_config
-      return @saml_config if @saml_config
+    def saml_config(idp_entity_id = nil)
+      return file_based_config if file_based_config
+      return adapter_based_config(idp_entity_id) if Devise.idp_settings_adapter
 
+      Devise.saml_config
+    end
+
+    private
+
+    def file_based_config
+      return @file_based_config if @file_based_config
       idp_config_path = "#{Rails.root}/config/idp.yml"
-      # Support 0.0.x-style configuration via a YAML file
+
       if File.exists?(idp_config_path)
-        Devise.saml_config = OneLogin::RubySaml::Settings.new(YAML.load(File.read(idp_config_path))[Rails.env])
+        @file_based_config ||= OneLogin::RubySaml::Settings.new(YAML.load(File.read(idp_config_path))[Rails.env])
+      end
+    end
+
+    def adapter_based_config(idp_entity_id)
```

Comment:
This method merges the adapter config settings into the existing `saml_config` so we don't lose any default settings.